### DISCUSSION TITLE
How should I use RustFmt to format my file?

### DISCUSSION THREAD

**Discussion by Catizard** [mrcjkb/rustaceanvim#172] [Q&A] [answered]
Hi, I'm just migrating from rust-tools to here, and I don't know how to use 'RustFmt'. Since when I typed it in a regular rs file and it shows me a pager screen which content is just the same file, and at the bottom it says
```
Error detected while processing function rustfmt#Format[7]..<SNR>49_RunRustfmt:                                                               
line   48:                                                                                                                                    
E776: No location list                                                                                                                        
Press ENTER or type command to continue  
```
Maybe I have a broken config, but I really don't know what's happening and how to solve this.
I tried searching at document and discussions and found nothing, so I opened this.

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

I think you've got the wrong plugin. This plugin doesn't define a `RustFmt` command.
[rust.vim](https://github.com/rust-lang/rust.vim) does.

To format your file with the LSP client started by this plugin, you can run 

`:lua vim.lsp.buf.format { async = true }`

I'd recommend a keymap:

```lua
vim.keymap.set({ 'n', 'v' }, '<space>f', function()
    vim.lsp.buf.format { async = true }
end, desc('[lsp] format buffer'))
```

which will format the entire buffer in normal mode, or just the visual selection in visual mode.

**Reply by Catizard** [reply]
Thank you for your reply.
I checked directly call `vim.lsp.buf. format` and it works fine.
Original problem might be some strange default behaviour of astronvim, sorry for this

**Reply by 9mm** [reply]
Man... this blew my mind. I started making an issue how RustFmt wasnt honoring nightly and I found this issue, and found out this isnt even a part of rustacean. its just a part of default nvim 🤯 it says its the rust filetype plugin.... when i look at the location of the docs for this, its just installed with nvim it seems.

I found another issue I think on how to enable nightly stuff
